### ISSUE TITLE
Implement HasAriaLabel interface in Select

### ISSUE THREAD

**Issue by web-padawan** [vaadin/flow-components#4711]
### Describe your motivation

Part of #2946
Part of vaadin/platform#3803

There is no Java APIs for setting `aria-label` or `aria-labelledby` in the `TextField`. Let's add them.

### Describe the solution you'd like

1. Implement `HasAriaLabel` interface in `TextFieldBase` class shared by 7 field components,
2. Add necessary overrides to set ARIA attributes on `this.focusElement` (the value button),
3. Add unit tests ensuring that the attribute value overrides the one from the web component.

**Comment by DiegoCardoso** [comment]
The component differs from other field components because it doesn't have an `input` field as the focus element, but a custom component with `role="button"` instead. And `aria-labelledby` is used to assign the label and the selected item or placeholder so the focus element has the proper name defined.

That makes it impossible to use `aria-label` since the specs define that only one should be used at any given time. I tried to check if it would be possible to use `aria-describedby` for the selected item/placeholder, but it makes the announcement a little worse, at least in VO/Safari:

![image](https://user-images.githubusercontent.com/262432/232716992-4d03f021-2ff2-4894-a15e-4561bf52dc2a.png)

> Size, list box pop up, button
> M

One alternative I was prototyping is to keep using the `aria-labelledby` in both cases when using the accessible name with string or for the element IDs used as labels:

### `setAriaLabel(String)`

When using `setAriaLabel` in the flow component (`accessibleName` in the web component), the component would:
- create a `<label>` element
- generate a unique id for it
- set the text content with the property value
- append it to a proper slot (probably a new `sr-label` slot)
- the new slot would be inside a `<div>` with a new `sr-only` class, which makes it invisible, but accessible to screen readers
- replace the default label id with the generated id

### `setAriaLabellbedby(String)`

When using `setAriaLabelledby` (`accessibleNameRef` in the web component), the component would simply replace the default label id with the custom one.

> **Note**
> Since the `aria-labelledby` is also used for the selected item and the placeholder, they would not be removed when a call to `setAriaLabel`/`setAriaLabelledby` is made. **Only the ids for the label/form-item label would be replaced**. That would not follow the criteria defined in https://github.com/vaadin/platform/issues/3803:
> > Setting a custom` aria-labelledby` through this API should probably remove the default `aria-labelledby`, if any, rather than append to it, in order to avoid unnecessary duplicate announcements.

**Comment by rolfsmeds** [comment]
That criterion is only about the `aria-labelledby` reference to the built-in or form-item label. As `aria-labelledby` is used for the value and placeholder as well in this case, it's not just OK but expected for those to remain there.
